src/main/kotlin/com/playwright/remote/engine/server/impl/ServerProvider.kt

```diff
@@ -0,0 +1,32 @@
+package com.playwright.remote.engine.server.impl
+
+import com.playwright.remote.core.enums.BrowserType
+import com.playwright.remote.core.enums.Platform
+import com.playwright.remote.engine.server.api.IServerProvider
+import java.nio.file.Paths
+
+class ServerProvider : IServerProvider {
+    private lateinit var process: Process
+
+    override fun launchServer(platform: Platform, browserType: BrowserType): String? {
+        val pb = ProcessBuilder().apply {
+            val driverPath = Paths.get(String.format("drivers/%s", platform.platformType))
+            val node = driverPath.resolve(platform.nodeProcess).toString()
+            val cli = driverPath.resolve("package/lib/cli/cli.js").toString()
+            command(node, cli, "launch-server", browserType.browserName)
+        }
+
+        process = pb.start()
+        return process.inputStream.bufferedReader().lineSequence().firstOrNull()
+
+    }
+
+    override fun stopServer() {
```

Comment:
I think it is better to return exit code here instead of unit to detect abnormal closure of the server and throw exceptions at this point.

So this method should return Int. and could be rewritten in a few ways. And definition of this method in interface should be changed to return Int.
Number one:
```
override fun stopServer(): Int = try {
            process.destroy()
            process.waitFor()
        } catch (e: InterruptedException) {
            throw RuntimeException(e.message, e.cause)
        }
```
And number two:
```
override fun stopServer(): Int = process.runCatching {
            destroy()
            waitFor()
        }.getOrThrow()
```

Reply:
Thanks, fixed